agent/systemd/notify.go

```diff
@@ -0,0 +1,33 @@
+package systemd
+
+import (
+	"errors"
+	"net"
+	"os"
+)
+
+var NotifyNoSocket = errors.New("No socket")
+
+// Notifier provides a method to send a message to systemd.
+type Notifier struct{}
+
+// Notify sends a message to the init daemon. It is common to ignore the error.
+func (n *Notifier) Notify(state string) error {
```

Comment:
This should get documented some place, maybe on https://www.consul.io/docs/agent/basics.html (Running and Stopping).